ws: Add option to remotectl to combine and validate PEM files

petervo opened the pull request:
Use it to support openshift's automatic certificate generation feature.

stefwalter commented:
I think this needs an integration test.

stefwalter commented on pkg/kubernetes/standalone/src/cockpit-kube-launch/main.go:
```diff
@@ -70,8 +70,12 @@ func setupCertificates() {
+	// and the check results without
+	exec.Command("/usr/sbin/remotectl", "certificate",
+		"/var/run/secrets/ws-certs.d/tls.crt",
+		"/var/run/secrets/ws-certs.d/tls.key").CombinedOutput()
```
But does it work to call --ensure along arguments as well? Just double checking.

petervo replied:
This will fail if valid cert files aren't present. It doesn't fall back to generating a self-signed one if it fails. Which i think makes sense. That's why ensure is a separate call below.

stefwalter replied:
Yes agree.

stefwalter commented on src/ws/remotectl-certificate.c:
```diff
@@ -149,12 +140,129 @@ ensure_certificate (const gchar *user,
+        goto out;
+    }
+
+  if (!g_file_set_contents (target_path, combined->str, combined->len, &error))
```
What happens if the target path already exists? Do we just overwrite it? That seems risky. If you've thought about this and decided for just overwriting it ... at the very least it needs documentation.

stefwalter replied:
Overwriting is fine. Will also be the case on certificate renewal. Lets just document it properly.

stefwalter commented on src/ws/remotectl-certificate.c:
```diff
@@ -149,12 +140,129 @@ ensure_certificate (const gchar *user,
+    {
+      g_message ("%s", error->message);
+      if (g_unlink (target_path) < 0)
+        g_message ("Failed to delete invalid certificate: %s", target_path);
```
Suggest g_strerror(errno) or %m here

petervo commented:
Added tests and updated the docs, I also swtiched to ensuring that the .cert files have a key present. But if we don't like that I can remove that commit and test.
